### PR TITLE
[RLlib] Update autoregressive actions example.

### LINKED ISSUE #44662: [RLlib]  Connectors API get_actions does not compute action_logp when actions are present.
### What happened + What you expected to happen

I'm working with an autoregressive model and trying to leverage the new API stack. Because of the nature of autoregressive models, I need to be sampling actions during the model's forward passes to use as inputs to later stages of the model. I figured this was fine because I can simply return those actions in the "actions" key along side the "action_dist_inputs" that were used to compute those actions. The RLModule documentation [here](https://docs.ray.io/en/latest/rllib/rllib-rlmodule.html#writing-custom-single-agent-rl-modules) seems to suggest that is totally fine and "action_logp" would be computed automatically later. However, when trying to train I get "action_logp" key errors when `PPOTorchLearner` attempts to compute the logp_ratio during `compute_loss_for_module`.

I believe the issue stems from  ray/rllib/connectors/module_to_env/get_actions.py where if "actions" are present in the data already there is an early return skipping any attempt compute "action_logp" values.

```python
    def _get_actions(self, data, sa_rl_module, explore):
        # Action have already been sampled -> Early out.
        if Columns.ACTIONS in data:
            return

        # ACTION_DIST_INPUTS field returned by `forward_exploration|inference()` ->
        # Create a new action distribution object.
        action_dist = None
        if Columns.ACTION_DIST_INPUTS in data:
            if explore:
                action_dist_class = sa_rl_module.get_exploration_action_dist_cls()
            else:
                action_dist_class = sa_rl_module.get_inference_action_dist_cls()
            action_dist = action_dist_class.from_logits(
                data[Columns.ACTION_DIST_INPUTS],
            )
            if not explore:
                action_dist = action_dist.to_deterministic()

            # Sample actions from the distribution.
            actions = action_dist.sample()
            data[Columns.ACTIONS] = actions

            # For convenience and if possible, compute action logp from distribution
            # and add to output.
            if Columns.ACTION_LOGP not in data:
                data[Columns.ACTION_LOGP] = action_dist.logp(actions)
```

My plan to work around this right now, is to manually compute the log_probs myself during the forward pass and add them to the output. Or better yet, try using a custom connector?

### Versions / Dependencies

This is present in the 2.10 release as well as the current master.
I'm running python 3.11 on Manjaro.

### Reproduction script

Well it's hard to build a quick example of this without building a custom module from scratch. I will look around through examples to see if I can find a quick way to reproduce.

### Issue Severity

Low: It annoys or frustrates me.

### PR DESCRIPTION
**Pull request by simonsays1980**
## Why are these changes needed?

The autoregressive actions example had an environment in which the agent ould cheat by looking only on the state when defining both actions, `a1` and `a2`. This PR proposes a new environment to test autoregressive actions modules in which the agent has to watch both the state and the action `a1` to define the action `a2` optimally. Rewards are based on the absolute negative deviance between the desired action for `a2` and its actual counterpart.

Furthermore, this PR introduces the `ValueFunctionAPI` for the `AutoregressiveActionsRLM` in the corresponding example which simplifies code and fixes actually an error due to the old `_compute_values` definition.

## Related issue number

Closes #44662 

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [x] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/core/models/configs.py [hunk 1]
```diff
@@ -160,6 +160,13 @@ class _MLPConfig(ModelConfig):
             "_" are allowed.
         output_layer_bias_initializer_config: Configuration to pass into the
             initializer defined in `output_layer_bias_initializer`.
+        clip_log_std: If the log std should be clipped by `log_std_clip_param`.
```
Comment (sven1977) [inline rllib/core/models/configs.py:163]: nit: I feel like this comment is confusing. We should write that clipping is only applied to those action distribution parameters that encode the log-std for a DiagGaussian action distribution. Any other node's output (or if there is no DiagGaussian) is not clipped.

Mentioning the value function makes it confusing.

File: rllib/algorithms/sac/sac_catalog.py [hunk 2]
```diff
@@ -187,6 +194,8 @@ def build_pi_head(self, framework: str) -> Model:
             hidden_layer_activation=self.pi_and_qf_head_activation,
             output_layer_dim=required_output_dim,
             output_layer_activation="linear",
+            clip_log_std=is_diag_gaussian,
+            log_std_clip_param=self._model_config_dict["log_std_clip_param"],
```
Comment (sven1977) [inline rllib/algorithms/sac/sac_catalog.py:198]: Should we do `.get` here to be defensive against any custom models that use custom model_config_dicts that are NOT derived from our gigantic (old) model config?

File: rllib/examples/envs/classes/correlated_actions_env.py [hunk 3]
```diff
@@ -1,44 +1,69 @@
...
+    Simple env in which the policy has to emit a tuple of correlated actions.
 
-    In each step, the agent observes a random number (0 or 1) and has to choose
-    two actions a1 and a2.
-    It gets +5 reward for matching a1 to the random obs and +5 for matching a2
-    to a1. I.e., +10 at most per step.
+    In each step, the agent observes a random number (between -1 and 1) and has
+    to choose two actions a1 and a2.
+
+    It gets 0 reward for matching a2 to the random obs times action a1. In all
```
[the remark this answers is not in the document]
Comment (sven1977) [reply rllib/examples/envs/classes/correlated_actions_env.py:14]: nit: Can we add a one-liner explaining that this reward structure makes sure that both actions must be correlated and that it's not possible for the network to learn each action head separately, only using the obs (like it was the case in the old example env)?

File: rllib/examples/rl_modules/autoregressive_actions_rl_module.py [hunk 4]
```diff
@@ -100,7 +102,7 @@
     # exceeds 150 in evaluation.
     stop = {
         f"{NUM_ENV_STEPS_SAMPLED_LIFETIME}": 100000,
-        f"{EVALUATION_RESULTS}/{ENV_RUNNER_RESULTS}/{EPISODE_RETURN_MEAN}": 150.0,
+        f"{EVALUATION_RESULTS}/{ENV_RUNNER_RESULTS}/{EPISODE_RETURN_MEAN}": -0.012,
```
Comment (sven1977) [inline rllib/examples/rl_modules/autoregressive_actions_rl_module.py:105]: nice!

Comment (sven1977) [reply rllib/examples/rl_modules/autoregressive_actions_rl_module.py:105]: Where does it roughly start?

File: rllib/examples/envs/classes/correlated_actions_env.py [hunk 5]
```diff
@@ -1,44 +1,69 @@
...
+        # Internal state for the environment (e.g., could represent a factor
+        # influencing the relationship)
+        self.state = None
+
+    def reset(self, seed=None):
+        """Reset the environment to an initial state."""
+        super().reset(seed=seed)
+
+        # Randomly initialize the state between -1 and 1
+        self.state = np.random.uniform(-1, 1, size=(1,))
```
Comment (sven1977) [inline rllib/examples/envs/classes/correlated_actions_env.py:41]: Nice that this can be negative, too. Makes sense!

File: rllib/examples/rl_modules/autoregressive_actions_rl_module.py (hunk 4 above)
Comment (simonsays1980) [reply rllib/examples/rl_modules/autoregressive_actions_rl_module.py:105]: It roughly starts at around `-0.55` - `-0.6`

Comment (simonsays1980) [reply rllib/examples/rl_modules/autoregressive_actions_rl_module.py:105]: ray_tune_evaluation_env_runners_agent_episode_returns_mean_default_agent

Comment (sven1977) [reply rllib/examples/rl_modules/autoregressive_actions_rl_module.py:105]: Niiice!!

**Review by sven1977 — APPROVED** [review]
LGTM! Thanks @simonsays1980 :)
